### ISSUE TITLE
Building against the recent GPipe: type mismatch in GPipe.WindowBits?

### ISSUE THREAD

**Issue by kindaro** [plredmond/GPipe-GLFW#35]
Hello! I am new to GPU programming. Trying to build `GPipe-Test` with `ghc-8.4.3` _(using Stack `lts-12.4`)_, I get the Haskell cube rotating just fine, after I manually fix a very little type mismatch.

### The error:

```
% cd ~/code/haskell/opengl/GPipe-Test/GPipe-Test
% stack build
...
    Configuring GPipe-GLFW-1.4.1.1...
    Preprocessing library for GPipe-GLFW-1.4.1.1..
    Building library for GPipe-GLFW-1.4.1.1..
...
    /home/kindaro/code/haskell/opengl/GPipe-GLFW/GPipe-GLFW/src/Graphics/GPipe/Context/GLFW/Format.hs:50:31: error:
        • Couldn't match expected type ‘Maybe Int’ with actual type ‘Int’
        • In the first argument of ‘WindowHint'RedBits’, namely ‘red’
          In the expression: WindowHint'RedBits red
          In the expression:
            [WindowHint'sRGBCapable sRGB, WindowHint'RedBits red,
             WindowHint'GreenBits green, WindowHint'BlueBits blue, ....]
       |
    50 |     , GLFW.WindowHint'RedBits red
       |                               ^^^
...
```

There are about 6 errors total, all in the same top level definition. They are easily fixed by wrapping the variables in the `Just` type constructor.

### How to observe the problem:

1. Clone:
* `GPipe-Test` at `a0e7e93`.
* `GPipe-GLFW` at `be97c1c`.
* `GPipe-Core` at `6cb83c6`.

2. Patch `GPipe-test/stack.yaml` with the following: _(Adjust paths to local git clones as appropriate.)_

```diff
 extra-deps:
-- GPipe-2.2
-- GPipe-GLFW-1.4.0
-resolver: nightly-2017-04-23
+- ../../GPipe-Core/GPipe-Core
+- ../../GPipe-GLFW/GPipe-GLFW
+# resolver: nightly-2017-04-23
+resolver: lts-12.4
+allow-newer: true
```

3. Run `stack build` in `GPipe-test`.

### I can:

Submit a patch. _(It is trivial.)_

* * *

Thank you for your work making Haskell cubes rotate!

**Comment by kindaro** [comment]
See patch here: https://github.com/kindaro/GPipe-GLFW/commit/fcd1340c5f7a6478d17733f1309bf5d6464a1524

**Comment by tobbebex** [comment]
Hi, I just merged pull request #34 that adressed this. Thanks anyway!
